api/v1alpha1/fenceagentsremediation_webhook_test.go

```diff
@@ -27,6 +29,35 @@ var _ = Describe("FenceAgentsRemediation Validation", func() {
 				Expect(err.Error()).To(ContainSubstring("unsupported fence agent: %s", invalidAgentName))
 			})
 		})
+
+		Context("with OutOfServiceTaint strategy", func() {
+			var outOfServiceStrategy *FenceAgentsRemediation
+
+			BeforeEach(func() {
+				orgValue := validation.IsOutOfServiceTaintSupported
+				DeferCleanup(func() { validation.IsOutOfServiceTaintSupported = orgValue })
+
+				outOfServiceStrategy = getFAR(validAgentName, OutOfServiceTaintRemediationStrategy)
+			})
+			When("out of service taint is supported", func() {
+				BeforeEach(func() {
+					validation.IsOutOfServiceTaintSupported = true
+				})
+				It("should be allowed", func() {
+					_, err := outOfServiceStrategy.ValidateCreate()
+					Expect(err).To(Succeed())
+				})
+			})
+			When("out of service taint is not supported", func() {
+				BeforeEach(func() {
+					validation.IsOutOfServiceTaintSupported = false
+				})
+				It("should be denied", func() {
+					_, err := outOfServiceStrategy.ValidateCreate()
+					Expect(err).To(MatchError(ContainSubstring(outOfServiceTaintUnsupportedMsg)))
```

Comment:
Now all of checks are changed to the one liner like the following:
```
-               When("agent name was not found ", func() {
-                       It("should be rejected", func() {
-                               farTemplate := getTestFARTemplate(invalidAgentName)
-                               _, err := farTemplate.ValidateCreate()
-                               Expect(err).To(HaveOccurred())
-                               Expect(err.Error()).To(ContainSubstring("unsupported fence agent: %s", invalidAgentName))
+                       When("agent name was not found ", func() {
+                               It("should be rejected", func() {
+                                       farTemplate := getTestFARTemplate(invalidAgentName)
+                                       Expect(farTemplate.ValidateCreate()).Error().To(MatchError(ContainSubstring("unsupported fence agent: %s", invalidAgentName)))
+                               })
```
The "Expect(err).To(HaveOccurred())" is removed because we can check both of checks with the one liner.